doc/specs/#885 - winrt Terminal Settings.md

```diff
@@ -0,0 +1,205 @@
+ how to expose relevant settings to a TerminalControl instance.
+
+`TerminalSettings` (which implements `IControlSettings` and `ICoreSettings`) will be moved to TerminalApp and act as a bridge that queries `AppSettings`. Thus,
+ when TermControl requests `IControlSettings` data, `TerminalSettings` responds by extracting the
+ relevant information from `AppSettings`. TerminalCore operates the same way with `ICoreSettings`.
+
+#### Settings UI: Modifying and Applying the Settings
+
+The Settings UI will also have a reference to the `AppSettings settings` from TerminalApp using a shared smart pointer
+as `settingsSource`. When the Settings UI is opened up, the Settings UI will also have its own `AppSettings settingsClone`
+that is a clone of TerminalApp's `AppSettings`.
+```c++
+settingsClone = settingsSource.Clone()
+```
+
+As the user navigates the Settings UI, the relevant contents of `settingsSource` will be retrieved and presented.
+As the user makes changes to the Settings UI, XAML will hold on to the changes that are made.
+When the user saves/applies the changes in the XAML, `settingsClone` will be updated to reflect those changes,
+then `settingsClone.ApplyChanges("settings.json", settingsSource)` is called. This compares the changes between
+`settingsClone` and `settingsSource`, then injects the changes (if any) to `settings.json`.
+
+As mentioned earlier, TerminalApp detects a change to "settings.json" to update its `AppSettings`. 
+ Since the above triggers a change to `settings.json`, TerminalApp will also update itself. When
+ something like this occurs, `settingsSource` will automatically be updated too.
+
+In the case that a user is simultaneously updating the settings file directly and the Settings UI,
+ `settingsSource` and `settingsClone` can be compared to ensure that the Settings UI, the TerminalApp,
+ and the settings files are all in sync.
+
+ **NOTE:** In the event that the user would want to export their current configuration, `ApplyChanges`
+ can be used to export the changes to a new file.
+
+## UI/UX Design
+
+N/A
+
+## Capabilities
+
+### Accessibility
+
+N/A
+
+### Security
+
+N/A
+
+### Reliability
+
+N/A
+
+### Compatibility
+
+N/A
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+### Deserialization
+
+After deserializing the settings, injecting the new json into settings.json
+ should not remove the existing comments or formatting.
+
+## Future considerations
+
+### Dropdown Customization
+
+Profiles are proposed to be represented in an `IObservableVector`. They could be represented as an
+ `IObservableMap` with the guid or profile name used as the key value. The main concern with representing
+ profiles as a map is that this loses the order for the dropdown menu.
+
+Once Dropdown Customization is introduced, `Settings` is expected to serialize the relevant JSON and expose
+ it to TerminalApp (probably as an `IObservableVector<DropdownMenuItem>`). Since this new object would

```

Comment:
I think you mean _deserialize_?

Again, this is just a case of saying "the settings model will grow to support new settings"

Reply:
this is the same reason we got rid of the other section, about xaml theming

Reply:
I'll disagree with you a bit here. I'm half-proposing a refactor. We'd be pivoting from an ObservableVector to an ObservableMap cuz I feel like that's the long-term design that makes more sense (or at least wanna know why I'm wrong).

Reply:
Just propose that in the spec right now then